### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by mwellick**

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -1,25 +1,78 @@
...
 
 class IntegerRange:
-    pass
+    def __init__(self,
+                 min_amount: int,
+                 max_amount: int) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __get__(self, instance: object, owner: type) -> None:
```
Comment (Bohdan-Salamakha) [inline app/main.py]: invalid typehint for "owner"
specify typehint for "instance"

File: app/main.py [hunk 2]
```diff
@@ -1,25 +1,78 @@
...
+        return getattr(instance, self.protected_name)
+
+    def __set__(self, instance: object, value: Any) -> None:
```
Comment (Bohdan-Salamakha) [inline app/main.py]: specify typehint for "instance"

File: app/main.py [hunk 3]
```diff
@@ -1,25 +1,78 @@
...
+        if not isinstance(value, int):
+            raise TypeError("Value should be integer")
+        if value < self.min_amount or value > self.max_amount:
+            raise ValueError(f"Value should not be less than {self.min_amount}"
+                             f" and greater than {self.max_amount}")
+        setattr(instance, self.protected_name, value)
+
+    def __set_name__(self, owner: type, name: str) -> None:
```
Comment (Bohdan-Salamakha) [inline app/main.py]: specify typehint for "owner"

File: app/main.py [hunk 4]
```diff
@@ -1,25 +1,78 @@
...
+    age = IntegerRange(min_amount=14, max_amount=60)
+    height = IntegerRange(min_amount=120, max_amount=220)
+    weight = IntegerRange(min_amount=50, max_amount=120)
 
 
 class Slide:
-    pass
+    def __init__(self,
+                 name: str,
+                 limitation_class: type[SlideLimitationValidator]) -> None:
```
Comment (Bohdan-Salamakha) [inline app/main.py]:
```suggestion
                 limitation_class: Type[SlideLimitationValidator]) -> None:
```

**Review by Bohdan-Salamakha — CHANGES_REQUESTED** [review]

**Review by EdAlekseiev — APPROVED** [review]
Good job

**Pull request closed without merging**